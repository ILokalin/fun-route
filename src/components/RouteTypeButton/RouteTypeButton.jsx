import React from 'react';
import PropTypes from 'prop-types';
import IconRound from 'components/IconsBtn/IconRound';
import IconRoute from 'components/IconsBtn/IconRoute';


const ROUTE_TYPE = { POLYLINE: 'polyline' },
      TEXT_POLYLINE_TYPE = 'Показать маршрут на карте',
      TEXT_MULTIROUTE_TYPE = 'Вернуть режим редактирования';

export default class extends React.PureComponent {
  constructor (props) {
    super(props);

    this.changeRoutType = this.changeRoutType.bind(this);
  }

  static propTypes = {
    routeType: PropTypes.oneOf(['polyline', 'multiRoute']).isRequired,
    onChangeRoutType: PropTypes.func.isRequired,
  }

  static defaultProps = {
    routeType: ROUTE_TYPE.POLYLINE,
    onChangeRoutType: () => false,
  }


  changeRoutType (event) {
    const { target: { checked } } = event;

    this.props.onChangeRoutType(checked);
  }
  

  render () {
    const { routeType } = this.props;

    const routeTypeState = (routeType === ROUTE_TYPE.POLYLINE) ? false : true;
    const routeTypeCaption = routeTypeState ? TEXT_MULTIROUTE_TYPE : TEXT_POLYLINE_TYPE;

    return (
      <div className="route-type">
        <IconRoute />
        <h3 className="route-type__caption">{ routeTypeCaption }</h3>
        <label className="route-type__label">
          <input  className="route-type__toggle"
                  type="checkbox"
                  name="routeType"
                  checked={ routeTypeState }
                  value="route"
                  onChange={ this.changeRoutType }
                  data-index={ this.props.index } />
          <span className="route-type__toggle-marker"></span>
          <IconRound />
        </label>
      </div>
      );
  }
}
