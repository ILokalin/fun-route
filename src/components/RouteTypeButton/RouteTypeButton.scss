.route-type {
  box-sizing: border-box;
  display: grid;
  position: relative;
  grid-template-columns: 2.1875rem calc(100% - 5.375rem) 3.1875rem;
  align-items: center;
  border: 1px solid $LightElement-color;
  border-radius: 0.25rem;
  background-color: $ContainerBg-color;
  padding-left: 1rem;
  width: 100%;
  height: 3.5rem;
  user-select: none;

  & .icon {
    width: 1.875rem;
    height: 1.875rem;
  }

  &:hover &__toggle::after {
    background-color: $ActiveElement-color;
  }


  &__caption {
    padding-right: 0.625rem;
    padding-left: 0.625rem;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap; 
    color: $WhiteElement-color;
  }


  &__toggle {
    cursor: pointer;
    width: 0px;
    height: 0px;
    overflow: hidden;
    clip: rect(0 0 0 0);

    &:before {
      position: absolute;
      top: 1.1875rem;
      border: 0.125rem solid $LightElement-color;
      border-radius: 1rem;
      box-shadow: 0 0.4375rem 1.25rem rgba(0, 0, 0, 0.07);
      width: 1.875rem;
      height: 0.75rem;
      content: '';
    }

    &:checked:before {
      background-color: $WhiteElement-color;
    }

    &:hover + .route-type__toggle-marker {
      transform: rotate(90deg);
      transition-duration: 0.3s;
      transition-timing-function: cubic-bezier(0.165, 0.84, 0.44, 1);
    }

    &:hover + .route-type__toggle-marker .icon__path {
      fill: $ActiveElement-color;
    }

    &:checked:hover + .route-type__toggle-marker .icon__path {
      fill: $LightElement-color;
    }

    &:checked + .route-type__toggle-marker {
      top: 0.75rem;
      right: 0.4375rem;
      background-color: $ActiveElement-color;
    }
  }


  &__toggle-marker {
    cursor: pointer;
    position: absolute;
    top: 0.75rem;
    right: 1.625rem;
    transition-duration: 0.3s;
    transition-timing-function: cubic-bezier(0.165, 0.84, 0.44, 1);
    border-radius: 50%;
    background-color: $ItemBg-color;
    width: 1.8125rem;
    height: 1.8125rem;

    & .icon__path {
      fill: $LightElement-color;
    }
  }
}

