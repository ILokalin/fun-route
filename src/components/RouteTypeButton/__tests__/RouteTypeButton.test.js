import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import rendrer from 'react-test-renderer';

import RouteTypeButton from 'components/RouteTypeButton'


describe('unit', () => {
  it('Route type button', () => {
    const component = rendrer.create(<RouteTypeButton />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    })
})
