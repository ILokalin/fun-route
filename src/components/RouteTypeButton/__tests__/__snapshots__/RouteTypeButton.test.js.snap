// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New point field Component create Route type button 1`] = `
<div
  className="route-type"
>
  <svg
    className="icon"
    fill="none"
    viewBox="0 0 30 30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="icon__path"
      d="M5.455 21.114c0-.334.27-.605.605-.605h.144a.605.605 0 110 1.21H6.06a.605.605 0 01-.605-.605zM18.572 8.372a2.016 2.016 0 00.641 3.924h5.482a2.016 2.016 0 00.642-3.924.605.605 0 00-.388 1.147.805.805 0 01-.255 1.567h-5.48a.805.805 0 01-.255-1.567.605.605 0 10-.387-1.147zM9.514 26.076a.605.605 0 10-.388 1.147.805.805 0 01-.254 1.566H3.392a.805.805 0 01-.255-1.567.605.605 0 10-.387-1.146A2.016 2.016 0 003.392 30h5.48a2.016 2.016 0 00.642-3.924zm-5.046-.482c-1.079-2.12-1.747-3.609-1.747-4.48 0-1.88 1.53-3.41 3.41-3.41 1.88 0 3.41 1.53 3.41 3.41 0 .872-.668 2.36-1.746 4.48 0 0-.386.761-.405.802-.156.326-.446.934-1.259.934s-1.102-.608-1.258-.934l-.405-.802zm-.536-4.48c0 .756 1.132 2.98 1.615 3.93 0 0 .515 1.032.55 1.073.02.03.053.014.07 0 .034-.041.091-.161.13-.242.023-.049.419-.83.419-.83.483-.95 1.615-3.174 1.615-3.93 0-1.214-.987-2.2-2.2-2.2s-2.2.986-2.2 2.2zM28.623.605v4.34a.605.605 0 01-.606.605h-5.458v2.79h.48a.605.605 0 110 1.21h-2.17a.605.605 0 110-1.21h.48V4.957v-.013V.618.605c0-.334.27-.605.605-.605h6.063c.334 0 .605.271.605.605zm-1.211.606h-4.853v3.128h4.853V1.211zm-3.734 13.184H6.321a3.738 3.738 0 01-3.733-3.734A3.738 3.738 0 016.32 6.927h12.206a.605.605 0 000-1.21H6.321a4.95 4.95 0 00-4.944 4.944 4.95 4.95 0 004.944 4.944h17.357a3.738 3.738 0 013.734 3.734 3.738 3.738 0 01-3.734 3.734H11.472a.605.605 0 100 1.21h12.206a4.95 4.95 0 004.944-4.944 4.95 4.95 0 00-4.944-4.944z"
      fill="#E9E9E9"
    />
  </svg>
  <h3
    className="route-type__caption"
  >
    Вернуть режим редактирования
  </h3>
  <label
    className="route-type__label"
    htmlFor="type-toggle"
  >
    <input
      className="route-type__toggle"
      id="type-toggle"
      name="routeType"
      onChange={[Function]}
      type="checkbox"
      value="route"
    />
    <span
      className="route-type__toggle-marker"
    />
    <svg
      className="icon"
      fill="none"
      viewBox="0 0 30 30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <g
        clipPath="url(#clip0)"
      >
        <path
          className="icon__path"
          d="M20.602 3.621l-1.516-.628a1.172 1.172 0 01.897-2.165l1.515.628a1.172 1.172 0 11-.896 2.165zM15 0a1.172 1.172 0 110 2.344C8.005 2.344 2.344 8.004 2.344 15c0 6.995 5.66 12.656 12.656 12.656 6.995 0 12.656-5.66 12.656-12.656A1.172 1.172 0 0130 15c0 8.29-6.709 15-15 15-8.29 0-15-6.709-15-15C0 6.71 6.709 0 15 0zM24.53 6.63l-1.16-1.16a1.172 1.172 0 111.656-1.657l1.16 1.16a1.172 1.172 0 11-1.657 1.658zM28.538 11.549a1.172 1.172 0 01-1.531-.634l-.628-1.516a1.172 1.172 0 112.165-.897l.628 1.516a1.172 1.172 0 01-.634 1.53z"
          fill="#798995"
        />
      </g>
      <defs>
        <clipPath
          id="clip0"
        >
          <path
            d="M0 0h30v30H0z"
            fill="#fff"
            transform="matrix(-1 0 0 1 30 0)"
          />
        </clipPath>
      </defs>
    </svg>
  </label>
</div>
`;
